igor-web/src/main/groovy/com/netflix/spinnaker/igor/jenkins/InfoController.groovy

```diff
@@ -73,11 +76,31 @@ class InfoController {
             throw new MasterNotFoundException("Master '${master}' does not exist")
         }
 
-        jenkinsClient.jobs.list.collect { it.name }
+        def jobList = []
+        def recursiveGetJobs
+
+        recursiveGetJobs = { list, prefix="" ->
+            if (prefix) {
+                prefix = prefix + "/job/"
+            }
+            list.each {
+                if (it.list == null || it.list.empty) {
+                    jobList << prefix + it.name
+                } else {
+                    recursiveGetJobs(it.list, prefix + it.name)
+                }
+            }
+        }
+        recursiveGetJobs(jenkinsClient.jobs.list)
+
+        return jobList
     }
 
-    @RequestMapping(value = '/jobs/{master}/{job:.+}')
-    JobConfig getJobConfig(@PathVariable String master, @PathVariable String job) {
+    @RequestMapping(value = '/jobs/{master}/**')
```

Comment:
In order to allow forward slashes in the names of Jenkins jobs, we can manually parse the URL like this. If this looks acceptable, I can do something similar in the `BuildController`. I'll have to move the job names to the end of all those URLs and manually parse them in this same way.